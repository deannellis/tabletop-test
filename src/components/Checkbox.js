import React, { Component, PropTypes } from 'react';

class Checkbox extends React.Component {
    state = {
        isChecked: false,
    }
    
    toggleCheckboxChange = () => {
        const { handleCheckboxChange } = this.props;
        const { isChecked } = this.state;
    
        this.setState(({ isChecked }) => (
            {
                isChecked: !isChecked,
            }
        ));
    
        handleCheckboxChange(!isChecked);
    }
    
      render() {
        const { label } = this.props;
        const { isChecked } = this.state;
        let isDisabled;
        if(isChecked) {
            isDisabled = false;
        } else {
            isDisabled = this.props.isDisabled
        }
    
        return (
          <div className="checkbox">
            <label>
                <input
                    type="checkbox"
                    value={isChecked}
                    checked={isChecked}
                    onChange={this.toggleCheckboxChange}
                    disabled={isDisabled}
                />
    
              {label}
            </label>
          </div>
        );
      }
}

export default Checkbox;