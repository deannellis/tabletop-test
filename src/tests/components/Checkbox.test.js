/* global jest expect */
import React from 'react';
import { shallow } from 'enzyme';
import Checkbox from '../../components/Checkbox';

test('should render checkbox with no props', () => {
    const wrapper = shallow(<Checkbox />);
    expect(wrapper).toMatchSnapshot();
});

test('should render checkbox with props', () => {
    const wrapper = shallow(<Checkbox label="Test Label" isDisabled={true} />);
    expect(wrapper).toMatchSnapshot();
});