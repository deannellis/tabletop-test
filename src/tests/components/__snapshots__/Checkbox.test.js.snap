// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render checkbox with no props 1`] = `
<div
  className="checkbox"
>
  <label>
    <input
      checked={false}
      onChange={[Function]}
      type="checkbox"
      value={false}
    />
  </label>
</div>
`;

exports[`should render checkbox with props 1`] = `
<div
  className="checkbox"
>
  <label>
    <input
      checked={false}
      disabled={true}
      onChange={[Function]}
      type="checkbox"
      value={false}
    />
    Test Label
  </label>
</div>
`;
